public/js/index.js

```diff
@@ -62,24 +63,12 @@ function handleViewClick(lesson) {
 }
 
 function handleClearBtn() {
-	formElement.mouseOver = false;
-	formElement.onmouseover = () => {
-		if (!this.mouseOver) {
-			this.mouseOver = true;
-			clearBtn.removeAttribute('hidden')
-		}
-	}
-	formElement.onmouseleave = () => {
-		if (this.mouseOver) {
-			if (lessonInput.value.length === 0) {
-				clearBtn.setAttribute('hidden', '');
-				this.mouseOver = false;
-			}
-			if (quill.root.innerHTML !== '<p><br></p>' && quill.root.innerHTML.length >= 8) {
-				clearBtn.removeAttribute('hidden');
-				this.mouseOver = false;
-			}
-		}
+	let textLessonContent = quill.root.innerHTML;
+	let textTitleContent = document.querySelector('.create-lesson-input')
+	if (textTitleContent.value.length > 0 || (textLessonContent.length >= 8 && textLessonContent !== '<p><br></p>')) {
+		clearBtn.removeAttribute('hidden');
+	} else {
+		clearBtn.setAttribute('hidden', '');
```

Comment:
This is how I redid the function after having a good read of the documentation you provided.

So, inside the function I defined two variables which hold the text of the two fields in the form.
The function checks those fields text whether empty or containing text and sets the hidden attribute of the clear button accordingly.

Re-doing the function with the event-delegation in mind also reduced a lot of unnecessary lines of code. Thank you for the documentation it was really helpful.